tests/loggen/loggen.c

```diff
@@ -21,956 +20,400 @@
-  if (res->tv_usec >= 1e6)
-    {
-      res->tv_sec++;
-      res->tv_usec -= USEC_PER_SEC;
-    }
-}
+  int str_len;
 
-static ssize_t
-write_chunk(send_data_t send_func, void *send_func_ud, void *buf, size_t buf_len)
-{
-  ssize_t rc;
-  size_t pos = 0;
+  if (read_from_file)
+    str_len = read_next_message_from_file(buffer, buffer_size, syslog_proto, thread_id);
+  else
+    str_len = generate_log_line(buffer, buffer_size, syslog_proto, thread_id, seq);
 
-  while (pos < buf_len)
-    {
-      rc = send_func(send_func_ud, buf + pos, buf_len - pos);
+  g_mutex_lock(message_counter_lock);
+  sent_messages_num++;
+  raw_message_length += str_len;
 
-      if (rc < 0)
-        return -1;
+  if (thread_stat_count && csv)
+    thread_stat_count[thread_id]+=1;
 
-      else if (rc == 0)
-        {
-          errno = ECONNABORTED;
-          return -1;
-        }
-      pos += rc;
-    }
-  return pos;
+  g_mutex_unlock(message_counter_lock);
+
+  return str_len;
 }
 
 static int
-parse_line(const char *line, char *host, char *program, char *pid, char **msg)
+enumerate_plugins(const gchar *plugin_path, GPtrArray *plugin_array, GOptionContext *ctx)
 {
-  const char *pos0;
-  const char *pos = line;
-  const char *end;
-  int space = skip_tokens;
-  int pid_len;
-
-  /* Find token */
-  while (space)
-    {
-      pos = strchr(pos, ' ');
-
-      if (!pos)
-        return -1;
-      pos++;
+  GDir *dir;
+  const gchar *fname;
+  int number_of_loaded_plugins = 0;
 
-      space --;
-    }
-
-  pos = strchr(pos, ':');
-  if (!pos)
-    return -1;
-
-  /* pid */
-  pos0 = pos;
-  if (*(--pos) == ']')
+  dir = g_dir_open(plugin_path, 0, NULL);
+  if (!dir)
     {
-      end = pos - 1;
-      while (*(--pos) != '[')
-        ;
-
-      pid_len = end -
-                pos; /* 'end' points to the last character of the pid string (not off by one), *pos = '[' -> pid length = end - pos*/
-      memcpy(pid, pos + 1, pid_len);
-      pid[pid_len] = '\0';
-    }
-  else
-    {
-      pid[0] = '\0';
-      ++pos; /* the 'pos' has been decreased in the condition (']'), reset it to the original position */
+      ERROR("unable to open plugin directory %s\n",plugin_path);
+      return 0;
     }
 
-  /* Program */
-  end = pos;
-  while (*(--pos) != ' ')
-    ;
-
-  memcpy(program, pos + 1, end - pos - 1);
-  program[end-pos-1] = '\0';
-
-  /* Host */
-  end = pos;
-  while (*(--pos) != ' ')
-    ;
-
-  memcpy(host, pos + 1, end - pos - 1);
-  host[end-pos-1] = '\0';
-
-  *msg = ((char *)pos0) + 2;
-
-  return 1;
-}
-
-static size_t
-_get_now_timestamp(char *stamp, gsize stamp_size)
-{
-  struct timeval now;
-  struct tm tm;
-
-  gettimeofday(&now, NULL);
-  localtime_r(&now.tv_sec, &tm);
-  return strftime(stamp, stamp_size, "%Y-%m-%dT%H:%M:%S", &tm);
-}
-
-static int
-gen_next_message(FILE *source, char *buf, int buflen)
-{
-  static int lineno = 0;
-  int linelen;
+  DEBUG("search for plugins in directory %s\n",plugin_path);
```

Comment:
cosmetic: I like spaces between parameters. Same comment for the ERROR line above.

Reply:
done